lib/spack/spack/cmd/python.py

```diff
@@ -48,10 +51,47 @@ def python(parser, args, unknown_args):
     if unknown_args:
         tty.die("Unknown arguments:", " ".join(unknown_args))
 
+    # If arguments or a command are provided, just use python
+    if args.python_args or args.python_command:
+        args.python_interpreter = "python"
```

Comment:
If the user specifies `-i` along with more arguments, or a `-c` option, I think an error should be raised.

Reply:
Gotcha, this makes sense.